Services/User/classes/class.ilUserDefinedData.php

```diff
@@ -84,9 +84,15 @@ public function get(string $a_field): string
         return $this->user_data[$a_field] ?? '';
     }
 
-    public function getAll(): array // Missing array type.
+    /**
+     * @return array<string, string>
+     */
+    public function getAll(): array
     {
-        return $this->user_data;
+        return array_map(
+            static fn ($value): string => $value ?? '',
```

Comment:
Similar to `get` I suggest transforming the values when retrieved by a consumer, instead of casting them when `__read` is called.